full-service/src/service/memo.rs

```diff
@@ -100,6 +111,15 @@ where
             return Ok(false);
         };
 
+        // validating the HMAC requires the receipient's subaddress
+        // view private key, so fetch the recipient subaddress_index
+        // of the txo, and fail if this is not available (orphaned txo)
+        let subaddress_index = match txo.subaddress_index {
+            Some(subaddress_index) => subaddress_index as u64,
+            // None => return Ok(false),
+            None => return Err(MemoServiceError::TxoOrphaned(txo_id_hex.to_string())),
+        };
```

Comment:
This will be a bit more elegantly written like this:
```suggestion
        let Some(subaddress_index) = txo.subaddress_index else {
            return Err(MemoServiceError::TxoOrphaned(txo_id_hex.to_string()));
        };
```

Reply:
did it that way because I wanted to cast to subaddress_index to u64 on assignment, but I've moved that to when used instead.  Also, meant to delete the commented out piece!